test/interface_basic_types.test.ts

```diff
@@ -140,23 +140,95 @@ test('Test for IDL \'unsigned long long\' type', async () => {
+    for (var i = 1; i < precision; i++) {
+      if (Math.random() > 0.5)
+        fraction += base ** i
+    }
+    fraction += base ** precision  // set the smallest bit
+
+    // for exponent part
+    // test all value of exponent part
+    // because exponent value -127 is used to express 0
+    // so this must start from -126.
+    for (var i = -126; i <= 127; i++) {
+      var float_value = fraction * (2 ** i)
+      expect(test_interface.floatMethod(float_value)).toBe(float_value);
+    }
+
+    // these two cases are beyond the range of floating-point
+    var float_value = fraction * (2 ** -127)
+    expect(test_interface.floatMethod(float_value) != float_value).toBe(true);
+    float_value = fraction * (2 ** 128)
+    expect(test_interface.floatMethod(float_value) != float_value).toBe(true);
+  }
+
+  var float_min = 2 ** (-149)
+  expect(test_interface.floatMethod(float_min)).toBe(float_min);
+  // the value beyond the range of exponent
+  expect(test_interface.floatMethod(float_min / 2) != float_min / 2).toBe(true)
+  expect(test_interface.floatMethod(float_min / 2) != float_min / 2).toBe(true)
+
+  // NaN == NaN returns false, thus we must use isNaN function
+  expect(isNaN(test_interface.floatMethod(NaN))).toBe(true);
+  expect(test_interface.floatMethod(Infinity)).toBe(Infinity);
+  expect(test_interface.floatMethod(-Infinity)).toBe(-Infinity);
+});
+
 test('Test for IDL \'double\' type', async () => {
   let test_interface = new bacardi.TestInterface();
 
   // The double type is a floating point numeric type that corresponds to the
   // set of finite double-precision 64 bit IEEE 754 floating point numbers.
+
   expect(test_interface.doubleMethod(0.123456789012345))
       .toBe(0.123456789012345);
 
-  // FIXME(zino): We should test for comparing single-precision floating point.
-
-  // FIXME(zino): We should check whether the type is restricted or
-  // unrestricted.
+  var base = 1 / 2
+  var precision = 52
+  for (var test_case = 0; test_case < 100; test_case++) {
+    // for fraction part
+    // create a random number that has 52 bits precision
+    var fraction = 1  // set the biggest bit.
+    for (var i = 1; i < precision; i++) {
+      if (Math.random() > 0.5)
+        fraction += base ** i
+    }
+    fraction += base ** precision  // set the smallest bit
+
+    // for exponent part
+    // test all value of exponent part
+    // because exponent value -1022 is used to express 0
+    // so this must start from -1022.
+    for (var i = -1022; i <= 1023; i += Math.ceil(Math.random() * 100)) {
+      var double_value = fraction * (2 ** i)
+      expect(test_interface.doubleMethod(double_value)).toBe(double_value);
+    }
+    // Note that the range of double and the range of number in typescript is
+    // the same, so we can not test the value out of the range.
+
+    // NaN == NaN returns false, thus we must use isNaN function
+    expect(isNaN(test_interface.doubleMethod(NaN))).toBe(true);
```

Comment:
IMHO, `float` and `double` SHOULD NOT take these special values such as `NaN`, `Infinity` and `-Infinity`.
They are only allowed in `unrestricted float` and `unrestricted double`.
Please see: https://heycam.github.io/webidl/#idl-unrestricted-float